File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> s;

        for (int i = 0; i < nums.size(); i++) {
            int num = nums[i];
            if (s.count(num)) {
                return true;
            }
            s.insert(num);
        }

        return false;
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-3.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        hashMap = { 0: 1 }
        prefixSum = 0
        count = 0

        for num in nums:
            prefixSum += num
            if prefixSum - k in hashMap:
                count += hashMap[prefixSum - k]
            hashMap[prefixSum] = hashMap.get(prefixSum, 0) + 1

        return count